src/lib/Event/ContentService.php

```diff
@@ -133,6 +133,10 @@ public function deleteContent(ContentInfo $contentInfo): iterable
             ? $beforeEvent->getLocations()
             : $this->innerService->deleteContent($contentInfo);
 
+        if (!is_array($locations)) {
+            $locations = iterator_to_array($locations);
+        }
```

Comment:
PHP 8.3, right? Then `iterator_to_array` can work on arrays by default too.
```suggestion
        $locations = iterator_to_array($locations);
```